department-of-veterans-affairs/caseflow
# Queue | Fix serializing request_issues count for ama appeals

### Description
Within `TasksController.index`, tasks for both legacy and AMA appeals are serialized using `TaskSerializer`, which was returning `issue_count` as `object.appeal.issues.count`. However, for Appeals, [`issues`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/appeal.rb#L33) returns a hash containing `request_issues` and `decision_issues`, so the `count` would always be 2. This fixes the count, returning `request_issues.count` for AMA appeals and `issues.count` for legacy.

### Acceptance Criteria
- [ ] Issue count is correct in task table views

### Testing Plan
1. Go to `/queue` as `BVASCASPER1`
2. Confirm 2 AMA appeals have issue counts != 2
3. Click in to case details view, confirm issue counts match

<details><summary>before</summary>
<img src="https://user-images.githubusercontent.com/8533004/45692411-b4ba6300-bb28-11e8-9be5-4f7128139e69.png">
</details>
<details><summary>after</summary>
<img src="https://user-images.githubusercontent.com/8533004/45692412-b4ba6300-bb28-11e8-8aad-0fca0aac216b.png">
</details>
